server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponseTests.java

```diff
@@ -81,34 +85,68 @@ public void testShareResponsesUsingMappingHash() throws Exception {
             inList.add(new FieldCapabilitiesIndexResponse(index, hashing, fieldCaps, canMatch));
         }
         Randomness.shuffle(inList);
-        final List<FieldCapabilitiesIndexResponse> serializedList;
-        try (BytesStreamOutput output = new BytesStreamOutput()) {
-            FieldCapabilitiesIndexResponse.writeList(output, inList);
-            try (
-                StreamInput in = new NamedWriteableAwareStreamInput(
-                    output.bytes().streamInput(),
-                    new NamedWriteableRegistry(Collections.emptyList())
-                )
-            ) {
-                serializedList = FieldCapabilitiesIndexResponse.readList(in);
+        // old versions
+        {
+            final Version oldVersion = VersionUtils.randomPreviousCompatibleVersion(random(), Version.V_8_2_0);
+            final List<FieldCapabilitiesIndexResponse> serializedList;
+            try (BytesStreamOutput output = new BytesStreamOutput()) {
+                output.setVersion(oldVersion);
+                FieldCapabilitiesIndexResponse.writeList(output, inList);
+                try (
+                    StreamInput in = new NamedWriteableAwareStreamInput(
```

Comment:
do we need to wrap the stream with  a namedwriteableaware stream input here? I see an empty list later

Reply:
Copy and paste :(